MIPs/mip-x-multichainapi.md

```diff
@@ -0,0 +1,241 @@
+
+The Multichain API and associated CAIP standards promise to enable the following benefits for developers:
+- Ability to interact with EVM and non-EVM decentralized networks through a consistent interface
+- Employ interface negotiation patterns to adopt novel wallet features, while gracefully degrading for wallets that may not yet support them
+- Ability to simultaneously interact with multiple networks
+- Be notified of updates across multiple networks
+- Elimination of excessive error handling involved with chain-switching across EVM networks
+
+This proposal also generally encourages a privacy preserving and more secure alternative to provider injection (ex. `window.<provider>`) to establish communication between sites and browser extension wallets. 
+
+## Usage Example
+
+### Request
+An example structure for a JSON-RPC request that a dapp would send to request authorization(s) from MetaMask:
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_createSession",
+  "params": {
+    "optionalScopes": {
+      "wallet:eip155": {
+        "methods": ["wallet_addEthereumChain"],
+      },
+      "eip155:1": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+      },
+      "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp": {
+        "methods": ["signAndSendTransaction","signAllTransactions","signMessage"],
+        "notifications": ["accountChanged"],
+      }
+      "bip122:000000000019d6689c085ae165831e93": {
+        ...
+      }
+    },
+    "scopedProperties": {
+      // Scope-keyed proposals or declarations
+      "eip155:1": {
+        "extension_foo": "bar"    
+      }
+    },
+  },
+}
+```
+
+### Response
+An example structure for the corresponding JSON-RPC response that a dapp would receive from the wallet:
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "sessionScopes": { 
+      "eip155:wallet": {
+          "methods": ["wallet_addEthereumChain"],
+          "accounts": ["eip155:wallet:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "eip155:1": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+        "accounts": ["eip155:1:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp": {
+        "methods": ["signAndSendTransaction","signAllTransactions","signMessage"],
+        "notifications": ["accountChanged"],
```

Comment:
```suggestion
        "notifications": [],
```

maybe we just leave this blank in case this mock notification name confuses someone from the solona space

Reply:
idk. nbd

Reply:
afaik, this notification example is one that some wallets, such as Phantom, do implement.

Reply:
On second thought. I'll remove to avoid potential confusion in case this notification isn't broadly adopted.